insights/parsers/cgroups.py

```diff
@@ -0,0 +1,105 @@
+"""
+Cgroups - File ``/proc/cgroups``
+================================
+
+This parser reads the content of ``/proc/cgroups``.
+This file shows the control groups information of system.
+
+Sample ``/proc/cgroups`` file::
+
+    #subsys_name	hierarchy	num_cgroups	enabled
+    cpuset	10	48	1
+    cpu	2	232	1
+    cpuacct	2	232	1
+    memory	5	232	1
+    devices	6	232	1
+    freezer	3	48	1
+    net_cls	4	48	1
+    blkio	9	232	1
+    perf_event	8	48	1
+    hugetlb	11	48	1
+    pids	7	232	1
+    net_prio	4	48	1
+
+
+Examples:
+
+    >>> i_cgroups.get_num_cgroups("memory")
+    232
+    >>> i_cgroups.is_subsys_enabled("memory")
+    True
+    >>> i_cgroups.data[0].get('hierarchy')
+    '10'
+
+"""
+
+from .. import parser, CommandParser
+from insights.specs import Specs
+from . import ParseException, parse_delimited_table
+
+
+@parser(Specs.cgroups)
+class Cgroups(CommandParser):
+    """
+    Class ``Cgroups`` parses the content of the ``/proc/cgroups``.
+
+    Attributes:
+        data (list): A list of the subsystem cgroup information
+
+    """
+
+    def parse_content(self, content):
+        # replace #subsys_name to subsys_name
+        content[0] = content[0].replace("#", '')
+        table = parse_delimited_table(content)
+        self.data = [dict((k.lower(), v) for (k, v) in item.items()) for item in table]
```

Comment:
I think it's OK to assign the result of `parse_delimited_table` to `self.data` directly.

This loop will retrieve all the dicts which have the same keys from the list and apply the `lower()` to them (the `keys` which are the same to all of the dicts) repeatedly. 
If there is any chance that the `keys` could be in `uppercase`, better `lower` them in just one time, e.g. `content[0] = content[0].lower()`, before the `parse_delimited_table`.

PS, do you intend to lower all the `subsys_name` instead of the `keys` like `subsys_name	hierarchy	num_cgroups	enabled`?

Reply:
@chenlizhong a list doesn't really seem to be the best way to store this info.  Why not use an attributed called `subsystems` containing the values as dict?  Both of your methods have to iterate through the list to check for names.  Also it would be better to use attribute names that better represent the data instead of a generic `data` name.